commit 8956b02aa53026e2040530be3b987b5cda55ca19
Author: Zbigniew JÄ™drzejewski-Szmek <zbyszek@in.waw.pl>
Date:   Sat Sep 10 12:07:51 2016 +0100

    yummy:patchX:shared-install--fix-set-default-with-empty-root---4118-.patch

shared/install: fix set-default with empty root (#4118)

    
    https://bugzilla.redhat.com/show_bug.cgi?id=1374371
    
    When root was empty or equal to "/", chroot_symlinks_same was called with
    root==NULL, and strjoina returned "", so the code thought both paths are equal
    even if they were not. Fix that by always providing a non-null first argument
    to strjoina.
    (cherry picked from commit ae9efab711e7478b4f035edd00824d518bcf0d3c)

diff --git a/src/shared/install.c b/src/shared/install.c
index cb2a2e7..cf1e834 100644
--- a/src/shared/install.c
+++ b/src/shared/install.c
@@ -403,6 +403,9 @@ static bool chroot_symlinks_same(const char *root, const char *wd, const char *a
         /* This will give incorrect results if the paths are relative and go outside
          * of the chroot. False negatives are possible. */
 
+        if (!root)
+                root = "/";
+
         a = strjoina(path_is_absolute(a) ? root : wd, "/", a);
         b = strjoina(path_is_absolute(b) ? root : wd, "/", b);
         return path_equal_or_files_same(a, b);
